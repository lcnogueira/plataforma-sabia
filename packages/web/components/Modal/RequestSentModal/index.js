import React from 'react';
import styled, { css } from 'styled-components';

const RequestSentModal = () => {
	return (
		<Modal>
			<div>
				<img
					src="/time-management-rafiki.svg"
					alt="Ilustração de um rapaz segurando um relógio e apoiado em uma ampulheta"
				/>
			</div>

			<InfosContainer>
				<p>Sua solicitação foi enviada</p>
				<span>
					Seu cadastro será analisado pela nossa equipe, em até 72h enviaremos uma
					resposta
				</span>
			</InfosContainer>
		</Modal>
	);
};

const Modal = styled.div`
	display: flex;
	max-width: 77rem;

	> div:first-child > img {
		width: 100%;
		height: 100%;
	}
`;

const InfosContainer = styled.div`
	${({ theme: { colors } }) => css`
		display: flex;
		flex-direction: column;

		margin-left: 6.8rem;
		margin-right: 6rem;

		> p {
			color: ${colors.secondary};
			font-size: 2.8rem;
			font-weight: 500;
			margin-bottom: 0.8rem;
			line-height: 3.3rem;
		}

		> span {
			font-size: 1.6rem;
			line-height: 2.4rem;
		}
	`}
`;

export default RequestSentModal;
