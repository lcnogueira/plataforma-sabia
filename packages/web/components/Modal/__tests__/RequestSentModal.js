import React from 'react';

import { render, screen } from 'test-utils';
import RequestSentModal from '../RequestSentModal';

describe('<RequestSentModal />', () => {
	it('should render correctly', () => {
		const { container } = render(<RequestSentModal />);

		expect(screen.getByRole('img', { src: /time-management-rafiki.svg/i })).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
