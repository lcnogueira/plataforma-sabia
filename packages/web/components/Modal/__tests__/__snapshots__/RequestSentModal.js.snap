// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestSentModal /> should render correctly 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 6.8rem;
  margin-right: 6rem;
}

.c2 > p {
  color: hsl(166,100%,34%);
  font-size: 2.8rem;
  font-weight: 500;
  margin-bottom: 0.8rem;
  line-height: 3.3rem;
}

.c2 > span {
  font-size: 1.6rem;
  line-height: 2.4rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 77rem;
}

.c0 > div:first-child {
  max-width: 26rem;
}

.c0 > div:first-child > img {
  width: 100%;
  height: 100%;
}

@media screen and (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .c0 > div:first-child {
    display: none;
  }

  .c0 .c1 {
    margin-left: 0;
    margin-right: 0;
  }
}

<div>
  <div
    class="c0"
  >
    <div>
      <img
        alt="Ilustração de um rapaz segurando um relógio e apoiado em uma ampulheta"
        src="/time-management-rafiki.svg"
      />
    </div>
    <div
      class="c1 c2"
    >
      <p>
        Sua solicitação foi enviada
      </p>
      <span>
        Seu cadastro será analisado pela nossa equipe, em até 72h enviaremos uma resposta
      </span>
    </div>
  </div>
</div>
`;
